Resources/Storage/OccupancyPlugins/WLCGAccountingJson.py

```diff
@@ -0,0 +1,121 @@
+"""
+  Defines the plugin to take storage space information given by WLCG Accounting Json
+  https://twiki.cern.ch/twiki/bin/view/LCG/AccountingTaskForce#Storage_Space_Accounting
+  https://twiki.cern.ch/twiki/pub/LCG/AccountingTaskForce/storage_service_v4.txt
+  https://docs.google.com/document/d/1yzCvKpxsbcQC5K9MyvXc-vBF1HGPBk4vhjw3MEXoXf8
+"""
+import json
+import gfal2  # pylint: disable=import-error
+import os
+import tempfile
+import shutil
+
+from DIRAC import gLogger, gConfig
+from DIRAC import S_OK, S_ERROR
+
+
+class WLCGAccountingJson(object):
+  """ .. class:: WLCGAccountingJson
+
+  Occupancy plugin to return the space information given by WLCG Accouting Json
+  """
+  def __init__(self, se):
+    self.se = se
+    self.log = se.log.getSubLogger('WLCGAccountingJson')
+    self.name = self.se.name
+
+  def getOccupancy(self, **kwargs):
+    """ Returns the space information given by LCG Accouting Json
+
+        :returns: S_OK with dict (keys: SpaceReservation, Total, Free)
+    """
+    occupancyLFN = kwargs['occupancyLFN']
+
+    if not occupancyLFN:
+      return S_ERROR("Failed to get occupancyLFN")
+
+    tmpDirName = tempfile.mkdtemp()
+    filePath = os.path.join(tmpDirName, os.path.basename(occupancyLFN))
+
+    for storage in self.se.storages:
+      try:
+        ctx = gfal2.creat_context()
+        params = ctx.transfer_parameters()
+        params.overwrite = True
+        res = storage.updateURL(occupancyLFN)
+        if not res['OK']:
+          return res
+        occupancyURL = res['Value']
+        ctx.filecopy(params, occupancyURL, 'file://' + filePath)
+
+      except gfal2.GError as e:
+        detailMsg = "Failed to copy file %s to destination url %s: [%d] %s" % (
+            occupancyURL, filePath, e.code, e.message)
+        self.log.debug("Exception while copying", detailMsg)
+        continue
+
+      else:
+        break
+
+    if not os.path.isfile(filePath):
+      return S_ERROR('No WLCGAccountingJson file of %s is downloaded.' % (self.name))
+
+    with open(filePath, 'r') as path:
+      occupancyDict = json.load(path)
+
+    # delete temp dir
+    shutil.rmtree(tmpDirName)
+
+    if 'storageservice' not in occupancyDict:
+      return S_ERROR('Could not find storageservice component in %s at %s' % (occupancyLFN, self.name))
+    storageService = occupancyDict['storageservice']
+
+    if 'storageshares' not in storageService:
+      return S_ERROR('Could not find storageshares component in %s at %s' % (occupancyLFN, self.name))
+    storageShares = occupancyDict['storageservice']['storageshares']
+
+    # get storageReservation
+    spaceReservation = self.se.options.get('SpaceReservation')
+    if not spaceReservation:
+      self.log.debug(
+          'Get SpaceToken in storage parameters instead of SpaceReservation because it is not defined in CS')
+      for storage in self.se.storages:
```

Comment:
If I understand, this is a special case for SRM ?
In principle, if the configuration is tidied and consistent, if there is a SpaceToken in  SRM, there should be a SpaceReservation in the SE. Maybe I should specify that in the doc. 
Do you think you need this special case ?

Reply:
Well, if we can be sure that everyone puts 'SpaceReservation' in the SE configuration when move from the legacy space token to the novel accounting json, then this would not be necessary.
I myself (ueda) was the one who tried json without knowing about 'SpaceReservation', and thus without having it in the CS. If it is only me, we can get rid of this part of the code.
(But I imagine some people keeping rather obsolete config without knowing newer stuff even when documented, eg. you might have seen someone still using 'ProtocolName'...)

Reply:
This plugin is able to discover the SpaceReservation from the json, so it's fine :)

Btw, ProtocolName is broken from v6r19,, so if it is still in used, and the person did not noticed, he probably does not need his SE :D

Reply:
I deleted the part to get spaceToken from storage.getParameters().